.devcontainer/launch.sh

```diff
@@ -59,20 +93,220 @@ parse_options() {
+    }
+
+    translate_local_envvars_to_shell_syntax() {
+        sed -r 's/\$\{localEnv:([^\:]*):?(.*)\}/${\1:-\2}/g'
+    }
+
+    inline_local_workspace_folder() {
+        sed "s@\${localWorkspaceFolder}@$(pwd)@g"
+    }
+
+    inline_container_workspace_folder() {
+        sed "s@\${containerWorkspaceFolder}@${WORKSPACE_FOLDER:-}@g"
+    }
+
+    inline_local_workspace_folder_basename() {
+        sed "s@\${localWorkspaceFolderBasename}@$(basename "$(pwd)")@g"
+    }
+
+    transform_to_one_line() {
+        tabs_to_spaces \
+      | trim_leading_whitespace \
+      | trim_trailing_whitespace \
+      | inline_local_workspace_folder \
+      | inline_container_workspace_folder \
+      | inline_local_workspace_folder_basename \
+      | translate_local_envvars_to_shell_syntax \
+      | tr -s '\n' '\t'
+    }
+
+    json_string() {
+        transform_to_one_line \
+      | grep -Po "\"$1\":\s*\"(.*)\"" \
+      | sed -r "s/.*\"$1\":[ ]*\"([^\"]*)\",?.*/\1/g" \
+      | no_empty_lines
+    }
+
+    json_map() {
+        transform_to_one_line \
+      | grep -Po "\"$1\":\s*{(.*?)\s*?}[^\"]" \
+      | strip_enclosing_chars '{' '}' \
+      | tr -s '\t' '\n' \
+      | json_map_key_to_shell_syntax \
+      | no_empty_lines
+    }
+
+    json_array() {
+        transform_to_one_line \
+      | grep -Po "\"$1\":\s*\[(.*?)\s*?\][^\"]" \
+      | strip_enclosing_chars '[' ']' \
+      | tr -s '\t' '\n' \
+      | sed -r 's/", "/"\n"/g' \
+      | sed -r 's/^(.*"),$/\1/' \
+      | no_empty_lines
+    }
+
+    # Read image
+    local DOCKER_IMAGE="$(
+        json_string "image" < "${path}/devcontainer.json"
+    )"
+
+    docker pull "$DOCKER_IMAGE"
+
+    # Read workspaceFolder
+    local WORKSPACE_FOLDER="$(
+        json_string "workspaceFolder" < "${path}/devcontainer.json"
+    )"
+
+    # Read remoteUser
+    local REMOTE_USER="$(
+        json_string "remoteUser" < "${path}/devcontainer.json"
+    )"
+
+    if test -z "${REMOTE_USER:-}"; then
+        REMOTE_USER="$(
+            # Read remoteUser from image metadata
+            read_docker_image_metadata \
+          | json_string "remoteUser"
+        )"
+    fi
+
+    # Read runArgs
+    local -a RUN_ARGS="($(
+        json_array "runArgs" < "${path}/devcontainer.json"
+    ))"
+
+    # Read initializeCommand
+    local -a INITIALIZE_COMMAND="($(
+        json_array "initializeCommand" < "${path}/devcontainer.json"
+    ))"
+
+    local -a ENV_VARS="($(
+        json_map "containerEnv" < "${path}/devcontainer.json" \
+      | sed -r 's/(.*)=(.*)/--env \1=\2/'
+    ))";
+
+    local -a MOUNTS="($(
+        tee < "${path}/devcontainer.json"   \
+            1>/dev/null                     \
+            >(json_array "mounts")          \
+            >(json_string "workspaceMount") \
+      | xargs -r -I% echo --mount '%'
+    ))"
+
+    # Update run args and env vars
+
+    # Don't pass -it if running in CI
+    if [ "${CI:-false}" != "true" ]; then
```

Comment:
suggestion: Letting bash evaluate the `true` / `false` builtins directly simplifies these expressions:

```suggestion
    if ! ${CI:-false}; then
```

Reply:
I'm doing the explicit string comparison because I am setting it to `true` in the workflow, but it was unclear to me whether other environments might set it to something else (`1|0`, `yes|no`, etc.) which would lead to `if ${CI:-false}` evaluating to e.g. `if 0;` and exiting with an error.

Reply:
Wouldn't it break anyway if using those alternate bools, since this explicitly checks for the string `"true"`?

Dunno about other envs, but we don't set this anywhere in CCCL, we usually check `$GITHUB_ACTIONS` to detect whether or not we're in CI.

Reply:
If another env set `CI=1`, then yes, it wouldn't pass this `if [ "${CI:-false}" != "true" ]` check, but it would also not exit with `$? != 0`

Reply:
Updated to check `if ${CI:-"false"}`